### PR TITLE
fix(checkbox): Add RequiredTrue validator for md-checkbox

### LINKED ISSUE #4383: template-driven md-checkbox doesn't properly update its validity
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
Template-driven `md-checkbox`es should properly handle directive validators placed on them.

#### What is the current behavior?
Currently only reactively defined `md-checkbox`es properly update their validity according to the validator functions on them. Template-driven `md-checkbox`es do not: once they become valid (or if they begin valid), they never update when they should become invalid.

#### What are the steps to reproduce?
See [this plunk](http://plnkr.co/edit/07Uw5ehN6pgXq2MZ2SVy?p=preview).

#### Which versions of Angular, Material, OS, browsers are affected?
all

### PR DESCRIPTION
**Pull request by tinayuangao**
PR #4395 with workaround
Closes #4383

### CODE REVIEW

File: src/lib/checkbox/checkbox-required-validator.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  NG_VALIDATORS,
+} from '@angular/forms';
+
+export const _MdCheckboxRequiredValidator = CheckboxRequiredValidator;
+
+export const MD_CHECKBOX_REQUIRED_VALIDATOR: Provider = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => MdCheckboxRequiredValidator),
+  multi: true
+  };
```
Comment (jelbourn) [inline src/lib/checkbox/checkbox-required-validator.ts]: -2 indent on closing brace

File: src/lib/checkbox/checkbox-required-validator.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+@Directive({
```
Comment (jelbourn) [inline src/lib/checkbox/checkbox-required-validator.ts]: Add a description for why this directive is necessary?

File: src/lib/checkbox/checkbox.spec.ts [hunk 3]
```diff
@@ -735,6 +736,41 @@ describe('MdCheckbox', () => {
...
+      fixture.detectChanges();
+
+      expect(checkboxInstance.checked).toBe(true);
+      expect(ngModel.valid).toBe(true);
+
+      inputElement.click();
+      fixture.detectChanges();
+
+      expect(checkboxInstance.checked).toBe(false);
+      expect(ngModel.valid).toBe(false);
```
Comment (jelbourn) [inline src/lib/checkbox/checkbox.spec.ts]: Is there a way to check if the error is specifically due to the required validator?

**Comment by tinayuangao** [comment]
Comments addressed. Also added `mat-` prefixes. Please take another look. Thanks!

**Comment by jelbourn** [comment]
Looks like there's just a lint error now (missing license header)

**Pull request merged**